lib/components/wallet_connect/approve_token_transfer.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:qubic_wallet/di.dart';
+import 'package:qubic_wallet/flutter_flow/theme_paddings.dart';
+import 'package:qubic_wallet/helpers/global_snack_bar.dart';
+import 'package:qubic_wallet/helpers/re_auth_dialog.dart';
+import 'package:qubic_wallet/helpers/sendTransaction.dart';
+import 'package:qubic_wallet/l10n/l10n.dart';
+import 'package:qubic_wallet/models/wallet_connect/approve_token_transfer_result.dart';
+import 'package:qubic_wallet/services/wallet_connect_service.dart';
+import 'package:qubic_wallet/stores/application_store.dart';
+import 'package:qubic_wallet/styles/edge_insets.dart';
+import 'package:qubic_wallet/styles/text_styles.dart';
+import 'package:qubic_wallet/styles/themed_controls.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+
+class ApproveTokenTransfer extends StatefulWidget {
+  final PairingMetadata? pairingMetadata;
+  final String? fromID;
+  final String? fromName;
+  final int amount;
+  final String? toID;
+  final String? nonce;
+  const ApproveTokenTransfer(
+      {super.key,
+      required this.pairingMetadata,
+      required this.fromID,
+      this.nonce,
+      required this.fromName,
+      required this.amount,
+      required this.toID});
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _ApproveTokenTransferState createState() => _ApproveTokenTransferState();
+}
+
+class _ApproveTokenTransferState extends State<ApproveTokenTransfer> {
+  final ApplicationStore appStore = getIt<ApplicationStore>();
+  final WalletConnectService wcService = getIt<WalletConnectService>();
+  bool hasAccepted = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  List<Widget> getButtons() {
+    final l10n = l10nOf(context);
+
+    return [
+      Expanded(
+          child: ThemedControls.transparentButtonBigWithChild(
+              //Reject button
+              child: Padding(
+                  padding: const EdgeInsets.all(ThemePaddings.smallPadding),
+                  child: Text(l10n.generalButtonCancel,
+                      style: TextStyles.transparentButtonText)),
+              onPressed: () {
+                Navigator.pop(context);
+              })),
+      ThemedControls.spacerHorizontalNormal(),
+      Expanded(
+          child: ThemedControls.primaryButtonBigWithChild(
+              //Accept button
+              onPressed: () async {
+                //Authenticate the user
+                if (mounted) {
+                  bool authenticated = await reAuthDialog(context);
+                  if (!authenticated) {
+                    if (mounted) {
+                      //required to remove the warning
+                      Navigator.pop(context);
+                    }
+                    return;
+                  }
+                }
+
+                //Get current tick
+                var transactionTick = appStore.currentTick;
```

Comment:
here the current tick needs to be fetched from the network and the default + value added (it's currently 5).
Please check the implementation in the submit trx form

Reply:
Changed it to current + 5 . Will have to update when tick offset is added in settings

Reply:
what was the change? I still see 20.
There is no settings to change the default value.

Please do:

int latestTick = await _apiService.getCurrentTick();
int targetTick = latestTick + defaultTargetTickType.value;

Reply:
Done.